PokemonAPI/PokemonAPI.BusinessLayer/Implementations/DomainServices/BattleService.cs

```diff
@@ -0,0 +1,111 @@
+                    .ThenInclude(x => x.Ability),
+                cancellationToken: cancellationToken
+            );
+
+            var secondPokemon = await _repository.Get(
+                filter: dbPokemon => dbPokemon.Id == secondPokemonId,
+                 include: source => source
+                    .Include(x => x.Abilities)
+                    .ThenInclude(x => x.Ability),
+                cancellationToken: cancellationToken
+            );
+
+            int GoesFirst(Pokemon firstPokemon, Pokemon secondPokemon) {
+                if (firstPokemon.BaseStats.Speed > secondPokemon.BaseStats.Speed)
+                {
+                    return 1;
+                }
+                else if (firstPokemon.BaseStats.Speed < secondPokemon.BaseStats.Speed)
+                {
+                    return 2;
+                }
+                else
+                {
+                    return 1;
+                }
+            }
+
+            int first = GoesFirst(firstPokemon, secondPokemon);
+            int firstPokeHealth = firstPokemon.BaseStats.HealthPoints;
+            int secondPokeHealth = secondPokemon.BaseStats.HealthPoints;
+            bool specialFirst = false;
+            bool specialSecond = false;
+
+            do
+            {
+                if (first == 1)
+                {
+                    firstPokeHealth = firstPokemon.BaseStats.SpecialAttack - secondPokemon.BaseStats.SpecialDefense;
+                    first = 2;
+                    specialFirst = true;
+                }
+                else
+                {
+                    secondPokeHealth = secondPokemon.BaseStats.SpecialDefense - firstPokemon.BaseStats.SpecialAttack;
+                    first = 1;
+                    specialSecond = true;
+                }
+            }
+            while (specialFirst && specialSecond && firstPokeHealth > 0 || secondPokeHealth > 0);
+
+            do
+            {
+                if (first == 1)
+                {   
+                        firstPokeHealth = firstPokemon.BaseStats.Attack - secondPokemon.BaseStats.Defense;
+                        first = 2;
+                }
+                else
+                {
+                        secondPokeHealth = secondPokemon.BaseStats.Defense - firstPokemon.BaseStats.Attack;
+                        first = 1;
+                }
+                   
+            }
+            while (firstPokeHealth > 0 || secondPokeHealth > 0);
+            
+          
+            if (firstPokemon.BaseStats.HealthPoints > secondPokemon.BaseStats.HealthPoints)
+            {
+                return new BattleResult(firstPokemon.Id, secondPokemon.Id);
+            }
+            else if (firstPokemon.BaseStats.HealthPoints < secondPokemon.BaseStats.HealthPoints)
+            {
+                return new BattleResult(secondPokemon.Id, firstPokemon.Id);
+            }
+            else
+            {
+                return null;
```

Comment:
Do return battle result but place both ids as null and add property Draw that is boolean and default false. (unless it's a draw right)

Id props should be **nullable**